packages/nest/test/test-app.ts

```diff
@@ -8,19 +7,27 @@ import {
   FeatureClient,
   OpenFeatureModule,
   StringFeatureFlag,
+  FeatureContextService,
 } from '../src';
-import { Client, EvaluationDetails, FlagValue, InMemoryProvider } from '@openfeature/server-sdk';
+import { OpenFeatureClient, EvaluationDetails, FlagValue, InMemoryProvider } from '@openfeature/server-sdk';
+import { OpenFeatureContextService } from '../src/feature.service';
+import { SharedAsyncLocalStorage } from '../src/async-storage';
 
 @Injectable()
 export class OpenFeatureTestService {
   constructor(
-    @FeatureClient() public defaultClient: Client,
-    @FeatureClient({ name: 'namedClient' }) public namedClient: Client,
+    @FeatureClient() public defaultClient: OpenFeatureClient,
+    @FeatureClient({ name: 'namedClient' }) public namedClient: OpenFeatureClient,
+    @FeatureContextService() public contextService: OpenFeatureContextService,
   ) {}
 
   public async serviceMethod(flag: EvaluationDetails<FlagValue>) {
     return flag.value;
   }
+
+  public async serviceMethodWithDynamicContext(flagKey: string): Promise<boolean> {
+    return this.defaultClient.getBooleanValue(flagKey, false, this.contextService.getContext());
```

Comment:
> My ideia here is that we don´t need to provide the injected context explicitly.

How would you do this? Merging the OF Client and the ContextService?

Reply:
> I was thinking about passing additional context info here. Some date might not be available at interceptor level. It would still work by using destructuring the context from the interceptor.

Where would you pass this data?

Reply:
I've opened a PR to this branch to give an example.
Here's it https://github.com/open-feature/js-sdk/pull/738